app/src/main/java/com/nutomic/syncthingandroid/service/EventProcessor.java

```diff
@@ -117,11 +118,9 @@ public void onEvent(Event event) {
                 );
                 break;
             case "FolderRejected":
-                onFolderRejected(
-                    (String) event.data.get("device"),          // deviceId
-                    (String) event.data.get("folder"),          // folderId
-                    (String) event.data.get("folderLabel")      // folderLabel
-                );
+                for (Map<String,String> m : (List<Map<String,String>>) event.data.get("added")) {
```

Comment:
On the phone, hence hard to check, but seems strange that rejected has an added field?